WorkloadManagementSystem/Client/Matcher.py

```diff
@@ -68,11 +68,11 @@ def selectJob( self, resourceDescription, credDict ):
     result = self.tqDB.matchAndGetJob( resourceDict, negativeCond = negativeCond )
 
     if not result['OK']:
-      return result
+      raise RuntimeError( result['Message'] )
     result = result['Value']
     if not result['matchFound']:
       self.log.info( "No match found" )
-      raise RuntimeError( "No match found" )
+      return {}
```

Comment:
No, this is not correct. In case "No match found" the returned result of the Matcher service must be S_ERROR('No match found'). So, you can still an exception here ( although this is counter the DIRAC coding convention ), but then in the MatcherHandler you should analyse the exception string or just return S_ERROR( str(rte) ). 
BTW, what was the reason to choose using exceptions in the Matcher class ?